### PR TITLE
network: add unittest for hits function

### PR DESCRIPTION
**Pull request by kaushik-rohit**
1) add unittests
2) fix hits function for slack

Changes proposed in this pull request:
- fix hits function in slack's network.py
- add unittests for hits

## Checks
- [x] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Make sure you have added required documentation
- [x] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Notify reviewers
@rohangoel96

### CODE REVIEW

File: lib/slack/analysis/network.py [hunk 1]
```diff
@@ -784,4 +784,4 @@ def identify_hubs_and_experts(log_dict, nicks, nick_same_list):
         if node_name in top_hub:
             message_graph.node[node_name]['shape'] = 'square'
 
-    return message_graph
+    return message_graph, top_hub, top_keyword_overlap, top_auth
```
Comment (prasadtalasila) [inline lib/slack/analysis/network.py:787]: @kaushik-rohit you are changing the function called by driver files. All the drivers would need updating as well.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]
Please update the driver files.

File: lib/slack/analysis/network.py (hunk 1 above)
Comment (prasadtalasila) [reply lib/slack/analysis/network.py:787]: @kaushik-rohit The driver files at the top level need updating for any breaking changes to functions introduced in the last four commits of the dev branch.

Comment (kaushik-rohit) [reply lib/slack/analysis/network.py:787]: Sir, this affects only the slack analysis and this is the only change to any function. I will update the slack driver.

Comment (kaushik-rohit) [reply lib/slack/analysis/network.py:787]: @prasadtalasila Infact the driver is correct and doesn't require any change.
https://github.com/prasadtalasila/IRCLogParser/blob/master/slack.py#L143 (Makes correct use of the function '''identify_hubs_and_experts'''
But the method in analysis folder doesn't return the correct results (which is fixed with this PR). https://github.com/prasadtalasila/IRCLogParser/blob/master/lib/slack/analysis/network.py#L787

Comment (prasadtalasila) [reply lib/slack/analysis/network.py:787]: I am confused. If driver is making calls to wrong function, how did the driver file run at all?

Comment (prasadtalasila) [reply lib/slack/analysis/network.py:787]: I see that we didn't use the values stored in `top_hub, top_keyword_overlap, top_auth` variables. That's probably why we got away without any errors.

Comment (kaushik-rohit) [reply lib/slack/analysis/network.py:787]: Yes, i assume that is the reason but even a usage as below
``` 
def func():
    return ret1 
ret1, ret2, ret3 = func() 
```
which seems to the previous case should have raised an error.

Comment (prasadtalasila) [reply lib/slack/analysis/network.py:787]: The python interpreter didn't. :(

**Review by prasadtalasila — APPROVED** [review]

**Pull request merged**